app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -78,9 +81,19 @@ def render_show_after_invalid
 
   def update_invalid_user
     current_user.second_factor_attempts_count += 1
-    # set time lock if max attempts reached
-    current_user.second_factor_locked_at = Time.zone.now if current_user.max_login_attempts?
-    current_user.save
+
+    UpdateUser.new(
+      user: current_user,
+      attributes: set_time_lock_if_max_attempts_reached
```

Comment:
we are still updating `second_factor_attempts_count` above.  If I remove Line 83, a bunch of specs fail (so it is well tested, yay!)

The code you have above works the same way as the current code. do you think it is more readable? In general I prefer longer conditional statements like we have in `set_time_lock_if_max_attempts_reached` , find them easier to parse than trailing conditionals